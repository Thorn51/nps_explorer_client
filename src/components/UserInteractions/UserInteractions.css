/* Colors */
:root {
  --darkGunmetal: #122932;
  --blackCoral: #576066;
  --queenBlue: #386fa4;
  --middleBlue: #80ced7;
  --mintCream: #f7fff7;
  --platinum: #e0e2e3;
}

/* User Interactions*/

.user_interactions {
  display: flex;
  flex-direction: column;
  width: 90%;
  margin: 5px auto 5px auto;
  text-align: right;
}
