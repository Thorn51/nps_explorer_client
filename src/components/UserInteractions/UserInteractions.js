import React, { useState, useContext } from "react";
import { Favorite } from "@material-ui/icons";
import "./UserInteractions.css";
import { GlobalContext } from "../../context/GlobalState";

export default function UserInteractions(props) {
  const [favoritePark, setFavorite] = useState(0);

  const { favorite, addFavorite } = useContext(GlobalContext);

  let id = favorite.length === 0 ? 0 : favorite.length + 1;

  const onClick = () => {
    setFavorite({
      id: id,
      selected: !favoritePark.selected,
      parkCode: props.parkCode
    });

    addFavorite(favoritePark);
  };

  console.log(favoritePark, favorite);

  const favoriteIcon =
    favoritePark.selected === true ? (
      <Favorite onClick={onClick} color="secondary" fontSize="large" />
    ) : (
      <Favorite onClick={onClick} fontSize="large" />
    );

  return <div className="user_interactions">{favoriteIcon}</div>;
}
