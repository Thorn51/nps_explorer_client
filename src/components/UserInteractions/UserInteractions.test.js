import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import UserInteractions from "./UserInteractions";

describe("<UserInteractions />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<UserInteractions />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders the UI as expected", () => {
    const tree = renderer.create(<UserInteractions />).toJSON();
    expect("tree").toMatchSnapshot();
  });
});
