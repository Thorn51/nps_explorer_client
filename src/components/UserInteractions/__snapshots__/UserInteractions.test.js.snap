// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInteractions /> Renders the UI as expected 1`] = `"tree"`;
